src/tests/PhysicsTest.cpp

```diff
@@ -148,888 +148,858 @@ struct PhysicsTest : Cr::TestSuite::Tester {
-            objWrapper0->setTranslation({0, boxHalfExtent * 2, 0});
-            objWrapper0->setMotionType(esp::physics::MotionType::KINEMATIC);
-
-            esp::physics::VelocityControl::ptr velCon =
-                objWrapper0->getVelocityControl();
-            velCon->controllingLinVel = true;
-            velCon->linVel = {0.2, 0, 0};
-
-            objWrapper1->setTranslation({0, boxHalfExtent * 5, 0});
-
-            while (physicsManager_->getWorldTime() < 3.0) {
-              // take single sub-steps for velocity control precision
-              physicsManager_->stepPhysics(-1);
-            }
-            CORRADE_COMPARE_AS((objWrapper0->getTranslation() -
-                                Magnum::Vector3{0.6, boxHalfExtent * 2, 0.0})
-                                   .length(),
-                               1.0e-3, Cr::TestSuite::Compare::LessOrEqual);
-            CORRADE_COMPARE_AS(
-                (objWrapper1->getTranslation() -
-                 Magnum::Vector3{0.559155, boxHalfExtent * 4, 0.0})
-                    .length(),
-                2.0e-3, Cr::TestSuite::Compare::LessOrEqual);
-          } break;
-        }
 
-        // reset the scene
-        rigidObjectManager_->removeAllObjects();
-        physicsManager_->reset();  // time=0
+  resetCreateRendererFlag(RendererEnabledData[testCaseInstanceId()].enabled);
+
+  std::string objectFile = Cr::Utility::Directory::join(
+      dataDir, "test_assets/objects/transform_box.glb");
+
+  std::string stageFile =
+      Cr::Utility::Directory::join(dataDir, "test_assets/scenes/plane.glb");
+
+  initStage(stageFile);
+
+  // ensure that changing default timestep does not affect results
+  physicsManager_->setTimestep(0.0041666666);
+
+  // We need dynamics to test this.
+  if (physicsManager_->getPhysicsSimulationLibrary() !=
+      PhysicsManager::PhysicsSimulationLibrary::NoPhysics) {
+    float boxHalfExtent = 0.2;
+
+    ObjectAttributes::ptr ObjectAttributes = ObjectAttributes::create();
+    ObjectAttributes->setRenderAssetHandle(objectFile);
+    ObjectAttributes->setBoundingBoxCollisions(true);
+    ObjectAttributes->setScale({boxHalfExtent, boxHalfExtent, boxHalfExtent});
+    auto objectAttributesManager =
+        metadataMediator_->getObjectAttributesManager();
+
+    int boxId =
+        objectAttributesManager->registerObject(ObjectAttributes, objectFile);
+    auto objTemplate = objectAttributesManager->getObjectByID(boxId);
+
+    auto& drawables = sceneManager_->getSceneGraph(sceneID_).getDrawables();
+
+    float stageCollisionMargin = 0.04;
+
+    for (int testId = 0; testId < 3; testId++) {
+      auto objWrapper0 =
+          makeObjectGetWrapper(objTemplate->getHandle(), &drawables);
+      auto objWrapper1 =
+          makeObjectGetWrapper(objTemplate->getHandle(), &drawables);
+
+      switch (testId) {
+        case 0: {
+          // test 0: stacking two DYNAMIC objects
```

Comment:
I'm still a bit on the fence about this, if it should be split into separate test cases (or test case instances), but since I can't figure out a simple enough solution to break those apart (without some UGLY lambdas), I guess it's best left as-is :) We can always improve upon this later.